app/scripts/components/criteria/variables.service.js

```diff
@@ -132,32 +129,32 @@ angular.module("chuvApp.components.criteria").factory("Variable", [
         return breadcrumb;
       });
 
-    resource.getVariableData = variableCode =>
-      resource.hierarchy().then(
-        data =>
-          new Promise(resolve => {
-            // find variable in the tree
-            const iterate = current => {
-              const children = current.groups || current.variables;
-              if (!children) {
-                return;
-              }
-
-              if (children.map(c => c.code).includes(variableCode)) {
-                const { code, label } = current;
-                const data = children.find(c => c.code === variableCode);
-                resolve({ data, parent: { code, label } });
-              }
+    resource.getVariableData = variableCode => {
+      return resource.hierarchy().then(data => {
+        let variableData = {}, found = false;
+        const iterate = current => {
+          const children = current.groups || current.variables;
+          if (!children) {
+            return;
+          }
 
```

Comment:
for...of requires also babel plugins. I could not get it running immediately, therefore I switched to an old school loop. 
But once we get support for for..of via plugins, this code can be updated and the test should still pass the same way as it does now.